Add candy and stardust costs for PvP

Pontiky opened the pull request:
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->


<!--
 Please make all PRs to the dev branch. The dev branch will be
 periodically pulled into the master branch. This allows time for
 the changes and documentation to be tested before being exposed to
 a wider population.
--->

## Description
<!-- In detail, describe what your PR adds to PokeAlarm -->
Added new DTS `<great_stardust>`, `<great_candy>`, `<ultra_stardust>`, `<ultra_candy>` to get the cost to power up a Pokemon for PvP to its final level/evolution/CP.
The candy cost is taking evolution and XL candies into account.

## Type of Change
<!-- Place a single 'x' into the correct box, ex: [x] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context
<!---
 Why is this change required? What problem does it solve?
 If it fixes an open issue, please link to the issue here.
-->

This change was requested and will close #728.
The new DTS can be used by players to know if it's worth to capture and power up a Pokemon for PvP.

## How Has This Been Tested?
<!---
 Please describe in detail how you tested your changes. Make sure to
 describe what tests you have performed, your testing environment,
 and if you have used this in a production setting. Please add
 screenshots if appropriate.
-->

I did multiple tests on a few Pokemons using PokeGenie app to compare values.
#### Test 1:
![image](https://user-images.githubusercontent.com/45545471/152661156-e0f8bd8e-b467-4465-96c1-d833d177777e.png)
<img src="https://user-images.githubusercontent.com/45545471/152661255-45ef3813-348d-4806-99cc-e09f6ee7973e.jpg" width="300"><img src="https://user-images.githubusercontent.com/45545471/152661310-4f951b92-82fb-4f36-9fa8-cee62362d340.jpg" width="300">

#### Test 2:
![image](https://user-images.githubusercontent.com/45545471/152661352-42bab87b-d1ab-47c7-ab3f-8b152e0b1b5e.png)
<img src="https://user-images.githubusercontent.com/45545471/152661337-ec27478f-e8e1-431b-aee7-83d742583471.jpg" width="300"><img src="https://user-images.githubusercontent.com/45545471/152661345-9c5a0259-1fb8-47d7-964e-a0655eb0f2ec.jpg" width="300">

#### Test 3:
![image](https://user-images.githubusercontent.com/45545471/152661434-ffc7144e-7207-4b72-b7cb-04ccab17a9bd.png)
<img src="https://user-images.githubusercontent.com/45545471/152661411-7dfa3729-a205-4c90-bcc1-85350987e2ba.jpg" width="300"><img src="https://user-images.githubusercontent.com/45545471/152661415-61f6b7ad-25ce-4f07-ab81-83fa46cfba39.jpg" width="300">


## Wiki Update
<!--
 Does this feature require an update to the wiki? If so, please submit
 the required change to https://github.com/RocketMap/PokeAlarmWiki.
 If your feature requires a wiki update, you may submit it for review
 but it will not be accepted until the wiki update is complete.
--->
- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

Included in this PR.

Pontiky committed:
added stardust and candy costs for PvP

Pontiky committed:
Added stardust and candy costs DTS

Pontiky committed:
Merge branch 'pokealarm:dev' into add-candy-stardust-costs-pvp

Gerbrum commented:
Working fine for me so far, just one small mistake I found:
`400: [130, 320, 333, 808, 714]` should be changed to `400: [129, 320, 333, 808, 714]` (Gyarados --> Magikarp)
I didn´t check all the evolution ID´s, saw the mistake as Magikarp is spawning right now.
Thanks for the great PR!

Pontiky committed:
updated evolution cost list

Pontiky committed:
Merge branch 'add-candy-stardust-costs-pvp' of https://github.com/Pontiky/pokealarm into add-candy-stardust-costs-pvp

Pontiky commented:
> Working fine for me so far, just one small mistake I found: `400: [130, 320, 333, 808, 714]` should be changed to `400: [129, 320, 333, 808, 714]` (Gyarados --> Magikarp) I didn´t check all the evolution ID´s, saw the mistake as Magikarp is spawning right now. Thanks for the great PR!

You're right! I replaced Gyrados by Magikarp.

I also updated the evolution cost list with missing Pokemons.

LegitDongo commented on PokeAlarm/Utilities/PvpUtils.py:
```diff
@@ -59,6 +59,101 @@ def min_level(limit, monster):
+
+def calculate_evolution_cost(monster_id, evolution_id, evolutions):
+    evolution_candy_costs = {
+        12: [10, 13, 16, 265, 293, 519, 664],
+        25: [1, 4, 7, 19, 29, 32, 41, 43, 60, 63, 66, 69, 74, 92, 111, 116,
+             133, 137, 147, 152, 155, 158, 161, 172, 173, 174, 175, 179, 183,
+             187, 220, 236, 238, 239, 240, 246, 252, 255, 258, 270, 273, 280,
+             287, 298, 304, 328, 355, 363, 371, 374, 387, 390, 393, 396, 403,
+             406, 439, 440, 443, 446, 495, 498, 501, 506, 535, 540, 543, 551,
+             574, 577, 582, 599, 602, 607, 610, 633, 650, 653, 656, 661, 669,
+             679, 704, 722, 725, 728, 731, 736, 761, 782, 789, 810, 813, 816,
+             821, 824, 837, 856, 859, 885],
+        50: [11, 14, 17, 21, 23, 25, 27, 35, 37, 39, 46, 48, 50, 52, 54, 56,
+             58, 72, 77, 79, 81, 83, 84, 86, 88, 90, 95, 96, 98, 100, 102,
+             104, 109, 113, 118, 120, 123, 138, 140, 163, 165, 167, 170, 177,
+             191, 194, 204, 209, 216, 218, 223, 228, 231, 261, 263, 266, 268,
+             276, 278, 283, 285, 290, 294, 296, 299, 300, 307, 309, 316, 318,
+             322, 325, 331, 339, 341, 343, 345, 347, 353, 360, 361, 366, 399,
+             400, 401, 408, 410, 412, 415, 418, 420, 422, 425, 427, 431, 433,
+             434, 436, 438, 447, 449, 451, 453, 456, 458, 459, 504, 509, 511,
+             513, 515, 517, 520, 522, 524, 527, 529, 532, 546, 548, 552, 554,
+             557, 559, 562, 564, 566, 568, 570, 572, 580, 585, 590, 592, 595,
+             597, 605, 613, 619, 622, 624, 627, 629, 659, 665, 667, 672, 677,
+             674, 682, 684, 686, 688, 690, 692, 694, 696, 698, 705, 712, 734,
+             742, 744, 747, 749, 751, 753, 755, 757, 759, 767, 769, 819, 827,
+             829, 831, 833, 835, 843, 846, 848, 850, 878],
+        100: [2, 5, 8, 33, 30, 42, 44, 61, 64, 67, 70, 75, 82, 93, 108, 112,
+              114, 117, 122, 125, 126, 148, 153, 156, 159, 176, 180, 188, 190,
+              193, 198, 200, 207, 215, 221, 233, 247, 253, 256, 259, 271, 274,
+              281, 288, 305, 315, 329, 349, 356, 364, 372, 375, 388, 391, 394,
+              397, 404, 430, 444, 478, 496, 499, 502, 507, 536, 541, 544, 575,
+              578, 583, 600, 603, 608, 611, 634, 651, 654, 657, 662, 670, 680,
+              723, 726, 729, 732, 737, 762, 783, 790, 811, 814, 817, 822, 825,
+              838, 857, 860, 886],
+        200: [525, 533, 588, 616, 710],
+        400: [129, 320, 333, 636, 714, 808]
```
This seems like data that should go in `data/base_stats.json`, since these are specific to the monster

Either that or in it's own data file; I just don't like having readily-changing data sitting plain in an app

What do you think @billyjbryant ?

Pontiky replied:
I thought about the same thing when I realized that there is a data folder with json files few hours ago. Maybe it could be a good idea.

billyjbryant replied:
I agree, either merged in to base-stats or its own data file in to data folder.

Pontiky replied:
I inserted the candy cost of evolutions in `base_stats.json` and modified the evo cost function consequently.

LegitDongo commented on PokeAlarm/Utilities/PvpUtils.py:
```diff
@@ -59,6 +59,101 @@ def min_level(limit, monster):
+def calculate_stardust_cost(start_level, target_level):
+    start_level = float(start_level)
+    target_level = float(target_level)
+    stardust_table = [200, 400, 600, 800, 1000, 1300, 1600, 1900, 2200,
+                      2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000,
+                      8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000]
```
Is this information subject to change in the future?

Pontiky replied:
Since the game release, it looks like they never changed the stardust and candy costs for powering up a Pokemon.
[Link to a reddit post, on July 2016](https://www.reddit.com/r/TheSilphRoad/comments/4sa4p5/stardust_costs_increase_every_4_power_ups/)

billyjbryant replied:
I would prefer to see data (even potentially long lived data) in a data file rather than hard coded into the programming.

Pontiky committed:
added evolution candy cost

Pontiky committed:
evolution cost calculated from base_stats

billyjbryant reviewed (changes requested):
Please move data values to a data file and out of the programming of the application.

Pontiky committed:
calculate costs from json
